src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java

```diff
@@ -42,7 +43,14 @@ public KafkaCallback(InflightMessageList inflightMessages, BinlogPosition positi
 	@Override
 	public void onCompletion(RecordMetadata md, Exception e) {
 		if ( e != null ) {
-			e.printStackTrace();
+			if ( e instanceof RecordTooLargeException ) {
+				LOGGER.error("RecordTooLargeException @ " + position + " -- " + key);
+				LOGGER.error("Maxwell dropped a row because it was too large (" + e.getLocalizedMessage() + ")");
+				LOGGER.error("Considering raising max.request.size broker-side.");
+				inflightMessages.completeMessage(position);
+			} else {
+				throw new RuntimeException(e);
```

Comment:
> I'm not sure what else could (reasonably) come up here.  Maybe `NotLeaderForPartition` -- that happens with `retries=0`... maybe a network timeout error could come up here, but I'm not sure what all we'd want to rescue and drop here.

I'm not sure how much the Java lib handles – in HC, there's a single exception that the sync producer will raise if messages cannot be delivered. I would err on the side of crashing, adding in failure handling as necessary. If you try it out on staging you'll get plenty chaos testing :-)

> I'm also thinking about the implications of duplicated maxwell messages in these crash-and-rewind type of scenarios.  For idempotent consumers, no big deal, but otherwise they'd really want to maintain a buffer of the last-N transactions, de-duping.  Or maybe I should just do the damn work and have maxwell read the last N transactions off each partition and de-dupe (like we talked about @dasch).

Yeah, you'd probably have to read _n_ messages, where _n_ is the batch size of the producer.